src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs

```diff
@@ -2,294 +2,446 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Collections.Generic;
 using Nethermind.Consensus.Messages;
 using Nethermind.Core;
-using Nethermind.Core.Crypto;
-using Nethermind.Core.Extensions;
 using Nethermind.Core.Specs;
+using Nethermind.TxPool;
+using Nethermind.Core.Crypto;
 using Nethermind.Crypto;
 using Nethermind.Evm;
 using Nethermind.Int256;
-using Nethermind.TxPool;
 
-namespace Nethermind.Consensus.Validators
+namespace Nethermind.Consensus.Validators;
+
+public sealed class TxValidator : ITxValidator
 {
-    public class TxValidator : ITxValidator
+    private readonly ITxValidator[] _validators;
+
+    public TxValidator(ulong chainId)
     {
-        private readonly ulong _chainIdValue;
+        _validators = new ITxValidator[byte.MaxValue + 1];
```

Comment:
Shouldn't we use the `Transaction.MaxTxType` introduced [here](https://github.com/NethermindEth/nethermind/pull/7334/files#diff-52dd146934935c1b5aa5650c8c5842ce450083c4f03ae076c0c1550745eeb750R22)

Reply:
Yeah we should

Reply:
Did it in my PR